src/partition/FileUtils.h

```diff
@@ -59,6 +65,35 @@ class InputFile {
     off_t _sz;
 };
 
+class InputFileArrow : public InputFile {
+public:
+    explicit InputFileArrow(boost::filesystem::path const &path, off_t blockSize);
+    ~InputFileArrow();
+
+    /// Return the size of the input file.
+    off_t size() const { return _sz; }
+
+    /// Return the path of the input file.
+    boost::filesystem::path const &path() const { return _path; }
+
+    int getBatchNumber() const;
+
+    /// Read a range of bytes into `buf`.
+    void read(void *buf, off_t off, size_t sz, int *bufferSize, std::vector<std::string> params) const;
+
+private:
+    // Disable copy construction and assignment.
+    InputFileArrow(InputFileArrow const &);
+    InputFileArrow &operator=(InputFileArrow const &);
+
+    boost::filesystem::path _path;
```

Comment:
I don't think this `_path` is set anywhere.

Reply:
@andy-slac it seems to be initialized in the c-tor of the class:
```
explicit InputFileArrow(boost::filesystem::path const &path, off_t blockSize);
```
Though, if the member `_path` is mutable then for the sake of safety it should be declared as:
```
boost::filesystem::path const _path;
```

Reply:
Done